src/tcgwars/logic/impl/gen4/PopSeries9.groovy

```diff
@@ -235,18 +255,23 @@ public enum PopSeries9 implements LogicCardInfo {
             energyCost C
             attackRequirement {}
             onAttack {
-              damage 0
+              flip {
+                discardRandomCardFromOpponentsHand()
+              }
             }
           }
           move "Nimble", {
             text "30 damage. If you have Turtwig in play, remove from Croagunk the number of damage counters equal to the damage you did to the Defending Pokémon."
             energyCost P, P
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+
+              if (my.all.findAll {it.name == "Turtwig" }) {
```

Comment:
You only need `find` or `any` here. These will both stop at finding one instead of collecting them all 

`find` will return the found value, `any` will return a boolean